internal/oci/container.go

```diff
@@ -835,6 +842,9 @@ func (c *Container) SetResources(s *specs.Spec) {
 
 // GetResources returns a copy of the Linux resources from Container.
 func (c *Container) GetResources() *types.ContainerResources {
+	c.metaLock.Lock()
+	defer c.metaLock.Unlock()
```

Comment:
https://github.com/cri-o/cri-o/pull/9531#discussion_r2465153989
Can't this be just read lock?